test/e2e-go/features/catchup/basicCatchup_test.go

```diff
@@ -98,20 +99,23 @@ func TestCatchupOverGossip(t *testing.T) {
 	partitiontest.PartitionTest(t)
 	defer fixtures.ShutdownSynchronizedTest(t)
 
-	t.Parallel()
-
 	syncTest := fixtures.SynchronizedTest(t)
 	supportedVersions := network.SupportedProtocolVersions
 	require.LessOrEqual(syncTest, len(supportedVersions), 3)
 
+	subTest := func(tt *testing.T, ledgerVer, fetcherVer string) {
+		tt.Run(fmt.Sprintf("ledger=%s,fetcher=%s", ledgerVer, fetcherVer),
+			func(t *testing.T) { runCatchupOverGossip(t, ledgerVer, fetcherVer) })
+	}
+
 	// ledger node upgraded version, fetcher node upgraded version
 	// Run with the default values. Instead of "", pass the default value
 	// to exercise loading it from the config file.
 	runCatchupOverGossip(syncTest, supportedVersions[0], supportedVersions[0])
 	for i := 1; i < len(supportedVersions); i++ {
-		runCatchupOverGossip(t, supportedVersions[i], "")
```

Comment:
did you run it locally? the original version had deterministic sequential execution order and the new one runs subtest in goroutines.

Reply:
my point was in even if not parallel, they run in non-deterministic order based on goroutime scheduling and hitting the test runner lock. Works then fine.

Reply:
`t.Run` source excerpt:

```golang
func (t *T) Run(name string, f func(t *T)) bool {
...
	// without being preempted, even when their parent is a parallel test. This
	// may especially reduce surprises if *parallel == 1.
	go tRunner(t, f)
	if !<-t.signal {
		// At this point, it is likely that FailNow was called on one of the
```

> Otherwise they will always run in order

They are not, I've hit this when was converting some catchpoint tests to subtest and it did not work because of the order test wanted.

Reply:
it looks blocked at `if !<-t.signal`. Although I do remember some ordering issues with subtest. Maybe parent was a parallel, do not remember :(